src/pollen.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Pollen charm business logic."""
+
+import os
+import shutil
+import time
+from glob import glob
+from subprocess import run
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+
+from exceptions import ConfigurationWriteError, InstallError
+
+
+def prepare() -> None:
+    """Install packages and write configuration files.
+
+    Raises:
+        InstallError: if the packages fail to install
+        ConfigurationWriteError: something went wrong writing the configuration
+    """
+    try:
+        apt.update()
+        apt.add_package(["pollen", "pollinate", "ent"])
+        # Apt-get makes the pollen restart error if we don't wait,
+        # even if apt-get executes correctly and there are no
+        # child processes left. The cause is unknown.
+        time.sleep(10)
+    except Exception as exc:
+        raise InstallError from exc
+    if not os.path.exists("/var/log/pollen"):
+        os.mkdir("/var/log/pollen")
+        shutil.chown("/var/log/pollen", "syslog")
+    try:
+        shutil.copy("files/usr.bin.pollen", "/etc/apparmor.d/usr.bin.pollen")
+        systemd.service_restart("apparmor.service")
+        run(["rsync", "files/logrotate.conf", "/etc/logrotate.d/pollen"], check=True)
+        run(["rsync", "files/rsyslog.conf", "/etc/rsyslog.d/40-pollen.conf"], check=True)
+        systemd.service_restart("rsyslog.service")
+    except Exception as exc:
```

Comment:
Same comment as before

Reply:
It is needed to restart rsyslog, the test broke when I tried to reload in this line.